src/main/java/gyro/google/GoogleStorageFileBackend.java

```diff
@@ -69,7 +71,34 @@ public String getCredentials() {
 
     @Override
     public Stream<String> list() throws Exception {
-        return null;
+        if (this.equals(GyroCore.getStateBackend(getName()))) {
+            List<StorageObject> storageObjects = new ArrayList<>();
+            Storage client = client();
+            String nextPageToken = null;
+            int count = 0;
+
+            do {
+                Objects objects = client.objects()
+                    .list(getBucket())
+                    .setPrefix(prefixed(""))
+                    .setPageToken(nextPageToken)
+                    .execute();
```

Comment:
This should be paginated just like we have a limit of 10k for the aws variant

Reply:
Addressed with Andrew's pr: #192